library/Zend/Form/Element.php

```diff
@@ -83,6 +86,20 @@ public function init()
     }
 
     /**
+     * Remove the password before rendering if the form fails in order to avoid any security issue
+     *
+     * @param  FormInterface $form
+     * @return mixed
+     * @see    \Zend\Form\Element\Password::prepareElement
+     */
+    public function prepareElement(FormInterface $form)
+    {
+        if ('password' == $this->getAttribute('type')) {
+            $this->setValue('');
```

Comment:
@Martin-P Actually, yes, I think that _would_ make more sense. There are cases where you want to keep the value of the password in the element, even if you do not want to _display_ it. As such, doing this in the view layer makes more sense, as display issues are view issues.